### PR TITLE
Improved the message for the case of Dockerhub not having v2 manifests.

### PR DESCRIPTION
**Pull request by facundobatista**
The previous error was a mess in the CLI; now the info is just logged and the message is nicer.

### CODE REVIEW

File: charmcraft/commands/store/registry.py [hunk 1]
```diff
@@ -176,8 +176,8 @@ def get_manifest(self, reference):
             response = self._hit('GET', url, headers=headers)
             result = assert_response_ok(response)
             if result.get('schemaVersion') != 2:
-                raise CommandError(
-                    "Manifest v2 requested but got something else: {}".format(result))
+                logger.debug("Got something else when asking for a v2 manifest: %s", result)
+                raise CommandError("Manifest v2 not found for the indicated reference.")
             logger.debug("Got the v2 manifest ok")
```
Comment (jameinel) [inline charmcraft/commands/store/registry.py:181]: would it be clearer to just have "not found for {}".format(ref)

rather than "indicated reference"?

**Review by jameinel — COMMENTED** [review]
thanks for the quick polish. I have not QAd this directly.

**Review by jameinel — APPROVED** [review]

File: charmcraft/commands/store/registry.py [hunk 2]
```diff
@@ -176,8 +176,8 @@ def get_manifest(self, reference):
             response = self._hit('GET', url, headers=headers)
             result = assert_response_ok(response)
             if result.get('schemaVersion') != 2:
-                raise CommandError(
-                    "Manifest v2 requested but got something else: {}".format(result))
+                logger.debug("Got something else when asking for a v2 manifest: %s", result)
+                raise CommandError("Manifest v2 not found for the indicated reference.")
```
Comment (sergiusens) [inline charmcraft/commands/store/registry.py:180]:
```suggestion
                raise CommandError("Manifest v2 not found for {reference!r}.")
```

**Review by sergiusens — APPROVED** [review]

**Review by cjp256 — APPROVED** [review]

File: charmcraft/commands/store/registry.py (hunk 1 above)
Comment (facundobatista) [reply charmcraft/commands/store/registry.py:181]: Indeed, thanks!

File: charmcraft/commands/store/registry.py (hunk 2 above)
Comment (facundobatista) [reply charmcraft/commands/store/registry.py:180]: Yes,

**Pull request merged**